service/tcp.go

```diff
@@ -49,7 +49,7 @@ func remoteIP(conn net.Conn) net.IP {
 }
 
 // Wrapper for logger.Debugf during TCP access key searches.
-func debugTCP(cipherID, template string, val interface{}) {
+func debugTCP[T any](cipherID, template string, val T) {
```

Comment:
Why?  In general, I think we will only add generics where doing so improves clarity and type safety or has a major performance benefit.